.github/workflows/unittests.yml

```diff
@@ -10,45 +10,54 @@
 
  jobs:
    unittests:
-     name: conda (${{ matrix.os }}, ${{ matrix.environment-file }})
+     name: micromamba unittests (${{ matrix.os }}, ${{ matrix.environment-file }})
      runs-on: ${{ matrix.os }}
      timeout-minutes: 30
      strategy:
        matrix:
          os: [macos-latest, ubuntu-latest, windows-latest]
-         environment-file: [ci/36-PYPI.yaml, ci/36-PYPI-PLUS.yaml, ci/36-GIT.yaml, ci/36-GIT-PLUS.yaml, ci/37-PYPI.yaml, ci/37-PYPI-PLUS.yaml, ci/37-GIT.yaml, ci/37-GIT-PLUS.yaml, ci/38-PYPI.yaml, ci/38-PYPI-PLUS.yaml, ci/38-GIT.yaml, ci/38-GIT-PLUS.yaml, ci/39-PYPI.yaml, ci/39-PYPI-PLUS.yaml, ci/39-GIT.yaml, ci/39-GIT-PLUS.yaml]
+         environment-file: [ci/36-STABLE.yaml, ci/36-STABLE-PLUS.yaml, ci/36-DEV.yaml, ci/36-DEV-PLUS.yaml, ci/37-STABLE.yaml, ci/37-STABLE-PLUS.yaml, ci/37-DEV.yaml, ci/37-DEV-PLUS.yaml, ci/38-STABLE.yaml, ci/38-STABLE-PLUS.yaml, ci/38-DEV.yaml, ci/38-DEV-PLUS.yaml, ci/39-STABLE.yaml, ci/39-STABLE-PLUS.yaml, ci/39-DEV.yaml, ci/39-DEV-PLUS.yaml]
          exclude:
-           - environment-file: ci/36-PYPI-PLUS.yaml
+           - environment-file: ci/36-STABLE-PLUS.yaml
              os: windows-latest
-           - environment-file: ci/36-GIT-PLUS.yaml
+           - environment-file: ci/36-DEV-PLUS.yaml
              os: windows-latest
-     defaults:
-       run:
-         shell: bash -l {0}
+           
      steps:
-       - uses: actions/checkout@v2
-       - uses: actions/cache@v2
-         env:
-           CACHE_NUMBER: 0
-         with:
-           path: ~/conda_pkgs_dir
-           key: ${{ matrix.os }}-conda-${{ env.CACHE_NUMBER }}-${{ hashFiles(matrix.environment-file) }}
-       - uses: conda-incubator/setup-miniconda@v2
+       - name: checkout repo
+         uses: actions/checkout@v2
+       
+       - name: setup micromamba
+         uses: mamba-org/provision-with-micromamba@main
          with:
-            miniconda-version: 'latest'
-            channels: conda-forge
-            channel-priority: strict
-            auto-update-conda: false
-            auto-activate-base: false
-            environment-file: ${{ matrix.environment-file }}
-            activate-environment: test
-            use-only-tar-bz2: true
-       - run: conda info --all
-       - run: conda list
-       - run: conda config --show-sources
-       - run: conda config --show
-       - run: pytest -v spaghetti --cov=spaghetti --doctest-modules --cov-config=.coveragerc --cov-report=xml
-       - name: codecov (${{ matrix.os }}, ${{ matrix.environment-file }})
+           environment-file: ${{ matrix.environment-file }}
+           micromamba-version: 'latest'
+      
+       - name: install bleeding edge libpysal - bash
+         shell: bash -l {0}
+         run: |
+           pip install pip
```

Comment:
This looks strange to my eyes. What does it do?

Reply:
It installs `pip`. Micromamba is a completely base installation, so I thought it's needed. Maybe it's not needed?

Reply:
Maybe it's better to put the `pip` requirement back in the `ci/*DEV.yaml` files and the we can cut the `pip install pip` line?

Reply:
How do you install pip with pip if you don't have pip? :)

> Maybe it's better to put the pip requirement back in the ci/*DEV.yaml

I think so.